### ISSUE TITLE
When will the code of COCO be released?

### ISSUE THREAD

**Issue by yangxinhaosmu** [CVI-SZU/CLIMS#11]
Very excellent work！Would you please release the code of COCO?

**Comment by Sierkinhane** [comment]
Hi, I will try to release the code for COCO in this month! 
Btw, welcome to star our project!

**Comment by yangxinhaosmu** [comment]
> Hi, I will try to release the code for COCO in this month! Btw, welcome to star our project!

I've stared it. Would you please release some code for COCO?

**Comment by Sierkinhane** [comment]
Hello, I will try to upload the code in these two days!

**Comment by Sierkinhane** [comment]
Sorry for the late update! Cause I'm busy in preparing an exam these days!

**Comment by Sierkinhane** [comment]
@yangxinhaosmu Hi, the code for training on COCO2014 has been uploaded! Feel free to ask any questions.

**Comment by yangxinhaosmu** [comment]
> @yangxinhaosmu Hi, the code for training on COCO2014 has been uploaded! Feel free to ask any questions.

Thanks！I have another question, I've found that the generated pseudo labels are all black. Is this situation resonable?

**Comment by Sierkinhane** [comment]
Hi, I  remember that the .png file stores the class index (0-21). So, the image looks dark.

**Comment by yangxinhaosmu** [comment]
> Hi, I remember that the .png file stores the class index (0-21). So, the image looks dark.

So, when I train deeplab, I should use the .png file directory as annotations directory. Is that right?

**Comment by Sierkinhane** [comment]
Exactly.

**Comment by yangxinhaosmu** [comment]
> Exactly.
Sorry to bother you, it maybe a simple question about deeplab results. When I train deeplab with train_aug.txt and test it with val.txt, how can I get the val results of segmentation? As we all know,  there are val results and test results. Thx

**Comment by Sierkinhane** [comment]
Hi, please follow the README of deeplab-pytorch.

python main.py test \
    --config-path configs/voc12.yaml \
    --model-path data/models/voc12/deeplabv2_resnet101_msc/train_aug/checkpoint_final.pth

The above script means to apply the trained deeplab on the CONFIG.DATASET.SPLIT.VAL(val set). If you want to generate predictions on test set, please change the [code](https://github.com/kazuto1011/deeplab-pytorch/blob/4219467fa5de07985f834f1bd8c04c186dc8f6d8/main.py#L317) to CONFIG.DATASET.SPLIT.TEST in main.py.

**Comment by Sierkinhane** [comment]
The results will be saved in `data/`, please check that!

**Comment by yangxinhaosmu** [comment]
> Hi, please follow the README of deeplab-pytorch.
> 
> python main.py test --config-path configs/voc12.yaml --model-path data/models/voc12/deeplabv2_resnet101_msc/train_aug/checkpoint_final.pth
> 
> The above script means to apply the trained deeplab on the CONFIG.DATASET.SPLIT.VAL(val set). If you want to generate predictions on test set, please change the [code](https://github.com/kazuto1011/deeplab-pytorch/blob/4219467fa5de07985f834f1bd8c04c186dc8f6d8/main.py#L317) to CONFIG.DATASET.SPLIT.TEST in main.py.

你好，但我发现JPEGImages文件夹里没有test.txt里对应的图像，JPEGImages里面是17125张图像对吧？

**Comment by yangxinhaosmu** [comment]
您用的test.txt也是
/JPEGImages/2008_000006.jpg
/JPEGImages/2008_000011.jpg
/JPEGImages/2008_000012.jpg
开头的吗

**Comment by Sierkinhane** [comment]
你可能需要去下载一下test set, 然后放到JPEGImages下

**Comment by Sierkinhane** [comment]
可以在PASCAL VOC官网上下载

**Comment by yangxinhaosmu** [comment]
> 你可能需要去下载一下test set, 然后放到JPEGImages下

我把voc的test set下载好，放到JPEGImages下了。但是在加载label_path的时候SegmentationClass里面没有对应的label，test set的label也可以在官网上下载吗？test的话应该是用voc而不是vocaug吧？

**Comment by yangxinhaosmu** [comment]
是要用deeplab那个库的demo.py来得到test的预测结果，然后上传到voc官网获得test的miou是吗？

**Comment by Sierkinhane** [comment]
Hi, test set是没有公开gt mask的，你需要保存你的预测结果 上传到Pascal voc的evaluation server中。

**Comment by yangxinhaosmu** [comment]
> Hi, test set是没有公开gt mask的，你需要保存你的预测结果 上传到Pascal voc的evaluation server中。你可以参照main.py中的test()函数来inference，简单改动一下就行。后面我会上传一下我的代码，不过你可以自己先尝试修改下。

谢谢同学，你的代码能救我的命！！

**Comment by Sierkinhane** [comment]
: -) 我尽快哈

**Comment by Sierkinhane** [comment]
@yangxinhaosmu Hi, I have uploaded the code for training deeplabv2. Please check that and feel free to ask any questions. :-)

**Comment by yangxinhaosmu** [comment]
> @yangxinhaosmu Hi, I have uploaded the code for training deeplabv2. Please check that and feel free to ask any questions. :-)

Your help is very much appreciated!!Thanks for your help!!!!

**Comment by yangxinhaosmu** [comment]
您好，请问在生成伪标签的阶段用cam_eval_thres是将CAM最小值设为0.15，然后再argmax吗？以及后面的0和255的判断还有些不懂，求指教Orz

**Comment by Sierkinhane** [comment]
你好，在eval_cam或者make_sem_label时，0.15是作为0通道背景的激活值，然后再沿着类别通道argmax获得语义分割标签。0是背景类，255是官方标签中一些去掉的区域（例如一些歧义的区域，无需计算loss）。
